# ##############################################################################
# Copyright (C) 2018-2024 Advanced Micro Devices, Inc. All rights Reserved.
#
# Permission is hereby granted, free of charge, to any person obtaining a copy
# of this software and associated documentation files (the "Software"), to deal
# in the Software without restriction, including without limitation the rights
# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
# copies of the Software, and to permit persons to whom the Software is
# furnished to do so, subject to the following conditions:
#
# The above copyright notice and this permission notice shall be included in all
# copies or substantial portions of the Software.
#
# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
# SOFTWARE.
#
# ##############################################################################
#
# Set a default for _LIBCUDACXX_HIP_TSC_CLOCKRATE and _LIBCUDACXX_HIP_TSC_NANOSECONDS_PER_CYCLE
# so that unknown architectures can be built
diff --git a/include/hip/std/detail/__config b/include/hip/std/detail/__config
index 7273edc8..aa596adf 100644
--- a/include/hip/std/detail/__config
+++ b/include/hip/std/detail/__config
@@ -122,6 +122,13 @@
 #elif defined(__gfx940__) || defined(__gfx941__) || defined(__gfx942__) || defined(__gfx1100__)
 #define _LIBCUDACXX_HIP_TSC_CLOCKRATE 100000000
 #define _LIBCUDACXX_HIP_TSC_NANOSECONDS_PER_CYCLE 10 // (1/_LIBCUDACXX_HIP_TSC_CLOCKRATE)
+#else
+#ifndef NDEBUG
+#warning Missing definition _LIBCUDACXX_HIP_TSC_CLOCKRATE and _LIBCUDACXX_HIP_TSC_NANOSECONDS_PER_CYCLE for architecture. Timing benchmark value will be meaningless.
+#endif
+// This should intentionally make no sense and tell the use that the value is meaningless
+#define _LIBCUDACXX_HIP_TSC_CLOCKRATE 1
+#define _LIBCUDACXX_HIP_TSC_NANOSECONDS_PER_CYCLE 1
 #endif

 #define _LIBCUDACXX_SYS_CLOCK_DURATION duration<long long, std::ratio<1,_LIBCUDACXX_HIP_TSC_CLOCKRATE>>
